crates/wit-component/src/printing.rs

```diff
@@ -1,15 +1,56 @@
 use anyhow::{anyhow, bail, Result};
+use std::borrow::Cow;
 use std::collections::HashMap;
-use std::fmt::{self, Write};
 use std::mem;
+use std::ops::Deref;
 use wit_parser::*;
 
 // NB: keep in sync with `crates/wit-parser/src/ast/lex.rs`
 const PRINT_F32_F64_DEFAULT: bool = true;
 
 /// A utility for printing WebAssembly interface definitions to a string.
 pub struct WitPrinter {
-    output: Output,
+    ext: WitPrinterExt<OutputToString>,
```

Comment:
FWIW it's ok to have breaking changes to this crate, so it might make more sense to repurpose the existing `WitPrinter` as `WitPrinter<O>` now?

Reply:
Sure.